polynote-kernel/src/main/scala/polynote/kernel/lang/python/PythonInterpreter.scala

```diff
@@ -524,7 +532,34 @@ class PythonInterpreter(val kernelContext: KernelContext, dependencyProvider: De
     cellContext: CellContext,
     code: String,
     pos: Int
-  ): IO[Option[Signatures]] = IO.pure(None)
+  ): IO[Option[Signatures]] = withJep {
+    try {
+      val (line, col) = populateCodeAndGlobals(cellContext, code, pos)
+      jep.eval(s"__polynote_sig__ = jedi.Interpreter(__polynote_code__, [__polynote_globals__, {}], line=$line, column=$col).call_signatures()[0]")
+      // If this comes back as a List, Jep will mash all the elements to strings. So gotta use it as a PyObject. Hope that gets fixed!
+      // TODO: will need some reusable PyObject wrappings anyway
+      val sig = jep.getValue("__polynote_sig__", classOf[PyObject])
+      val index = sig.getAttr("index", classOf[java.lang.Long])
+      val params = sig.getAttr("params", classOf[Array[PyObject]]).map {
+        paramObj => paramObj.getAttr("name", classOf[String])
+      }
+      val docString = Try(sig.getAttr("docstring", classOf[PyCallable]).call())
+        .map(_.asInstanceOf[String])
+        .toOption.filterNot(_.isEmpty).map(ShortString.truncate)
+
+      val name = s"${sig.getAttr("name", classOf[String])}(${params.mkString(", ")})"
+      val hints = ParameterHints(
+        name,
+        docString,
+        params.toList.map {
+          name => ParameterHint(name, "", None) // TODO: can individual params have docstrings?
```

Comment:
Hm, can we at least get the type of the param to show?

Reply:
What type? 😛 This is python.

I couldn't figure out how to get info from jedi about type hints if there are any, but I'll play around with it a bit.

Reply:
Turns out that after upgrading jedi, you can get type names through the `infer` method (for some reason?). Added that in.

Reply:
Nice!